diff --git a/src/systems/camera-system.js b/src/systems/camera-system.js
index caa0421744..6457a84d18 100644
--- a/src/systems/camera-system.js
+++ b/src/systems/camera-system.js
@@ -158,6 +158,7 @@ export const CAMERA_MODE_THIRD_PERSON_NEAR = 1;
 export const CAMERA_MODE_THIRD_PERSON_FAR = 2;
 export const CAMERA_MODE_INSPECT = 3;
 export const CAMERA_MODE_SCENE_PREVIEW = 4;
+export const CAMERA_MODE_THIRD_PERSON_VIEW = 5;

 const NEXT_MODES = {
   [CAMERA_MODE_FIRST_PERSON]: CAMERA_MODE_THIRD_PERSON_NEAR,
@@ -209,6 +210,7 @@ export class CameraSystem {
     if (customFOV) {
       this.viewingCamera.fov = customFOV;
     }
+
     this.viewingCamera.layers.enable(Layers.CAMERA_LAYER_VIDEO_TEXTURE_TARGET);
     this.viewingCamera.layers.enable(Layers.CAMERA_LAYER_FIRST_PERSON_ONLY);
     this.viewingCamera.layers.enable(Layers.CAMERA_LAYER_UI);
@@ -255,6 +257,22 @@ export class CameraSystem {
     this.mode = NEXT_MODES[this.mode] || 0;
   }

+  setMode(cameraMode) {
+    if (cameraMode > CAMERA_MODE_THIRD_PERSON_VIEW || cameraMode < 0 || cameraMode == this.mode) return;
+    const vrMode = AFRAME.scenes[0].is("vr-mode") || AFRAME.utils.device.isMobileVR();
+    const mode = vrMode ? CAMERA_MODE_FIRST_PERSON : cameraMode; // do not change mode, if user is in VR mode
+
+    this.mode = mode;
+
+    if (this.mode == CAMERA_MODE_THIRD_PERSON_VIEW) {
+      this.viewingCamera.layers.disable(Layers.CAMERA_LAYER_FIRST_PERSON_ONLY);
+      this.viewingCamera.layers.enable(Layers.CAMERA_LAYER_THIRD_PERSON_ONLY);
+    } else {
+      this.viewingCamera.layers.disable(Layers.CAMERA_LAYER_THIRD_PERSON_ONLY);
+      this.viewingCamera.layers.enable(Layers.CAMERA_LAYER_FIRST_PERSON_ONLY);
+    }
+  }
+
   inspect(el, distanceMod, fireChangeEvent = true) {
     const { inspectable, pivot } = getInspectableAndPivot(el);

@@ -423,7 +441,8 @@ export class CameraSystem {
       }
       if (!this.enteredScene && entered) {
         this.enteredScene = true;
-        this.mode = CAMERA_MODE_FIRST_PERSON;
+        const thirdPersonEnabled = window.APP.store.state.preferences.enableThirdPersonView;
+        this.setMode(thirdPersonEnabled ? CAMERA_MODE_THIRD_PERSON_VIEW : CAMERA_MODE_FIRST_PERSON);
       }
       this.avatarPOVRotator = this.avatarPOVRotator || this.avatarPOV.components["pitch-yaw-rotator"];
       this.viewingCameraRotator = this.viewingCameraRotator || this.viewingCamera.el.components["pitch-yaw-rotator"];
@@ -521,6 +540,24 @@ export class CameraSystem {
             panY
           );
         }
+      } else if (this.mode === CAMERA_MODE_THIRD_PERSON_VIEW) {
+        this.viewingCameraRotator.on = false;
+        translation.makeTranslation(0, 0, 1);
+        this.avatarRig.object3D.updateMatrices();
+        setMatrixWorld(this.viewingRig.object3D, this.avatarRig.object3D.matrixWorld);
+        if (scene.is("vr-mode")) {
+          this.viewingCamera.updateMatrices();
+          setMatrixWorld(this.avatarPOV.object3D, this.viewingCamera.matrixWorld);
+        } else {
+          this.avatarPOV.object3D.updateMatrices();
+          setMatrixWorld(this.viewingCamera, this.avatarPOV.object3D.matrixWorld.multiply(translation));
+        }
+
+        this.avatarRig.object3D.updateMatrices();
+        this.viewingRig.object3D.matrixWorld.copy(this.avatarRig.object3D.matrixWorld);
+        setMatrixWorld(this.viewingRig.object3D, this.viewingRig.object3D.matrixWorld);
+        this.avatarPOV.object3D.quaternion.copy(this.viewingCamera.quaternion);
+        this.avatarPOV.object3D.matrixNeedsUpdate = true;
       }
     };
   })();
